### PR TITLE
[Core] Fix issue 15051 for msi login issue in azure.core

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**  

Fix issue #15051 

With azure.core, all get_token method will need two addition arguments: *scope and **kwargs.
Here is a fix for msi login.

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by jiasli** [comment]
1. Please follow the [PR title convention](https://github.com/Azure/azure-cli/blob/1cf3072bb2e3af3db4f9b3dffc575c7e2c46efc9/doc/authoring_command_modules/README.md#submitting-pull-requests) and describe what the "issue" is:
   > [Core] Fix #15051: `MSIAuthenticationWrapper` not working with `azure.core`

2. I can't seem to see the relationship with #15051

**Comment by yonzhan** [comment]
Core

**Review by jsntcy — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/adal_authentication.py [hunk 1]
```diff
@@ -90,6 +90,6 @@ def _log_hostname():
 
 class MSIAuthenticationWrapper(MSIAuthentication):
     # This method is exposed for Azure Core.
-    def get_token(self):
+    def get_token(self, *scopes, **kwargs):
```
Comment (arrownj) [inline src/azure-cli-core/azure/cli/core/adal_authentication.py:93]: It's strange that there is an argument named `*scopes` other than `*args`.

**Review by arrownj — APPROVED** [review]

**Comment by Juliehzl** [comment]
> 1. Please follow the [PR title convention](https://github.com/Azure/azure-cli/blob/1cf3072bb2e3af3db4f9b3dffc575c7e2c46efc9/doc/authoring_command_modules/README.md#submitting-pull-requests) and describe what the "issue" is:
>    > [Core] Fix #15051: `MSIAuthenticationWrapper` not working with `azure.core`
> 2. I can't seem to see the relationship with #15051

updated

**Pull request merged**